### PR TITLE
add multilabel option

### PR DESCRIPTION
**Pull request by justusschock**
## What does this PR do?
Adds multilabel option to classification tasks

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

cc @aribornstein

### CODE REVIEW

**Comment by kaushikb11** [comment]
@justusschock Could you add a test for it?

File: flash/core/classification.py [hunk 1]
```diff
@@ -32,6 +34,43 @@ def after_uncollate(self, samples: Any) -> Any:
...
 class ClassificationTask(Task):
 
+    def __init__(
+        self,
+        num_classes: int,
+        model: Optional[torch.nn.Module] = None,
+        loss_fn: Optional[Callable] = None,
+        multilabel: bool = False,
+        optimizer: Type[torch.optim.Optimizer] = torch.optim.SGD,
+        metrics: Union[pl.metrics.Metric, Mapping, Sequence, None] = None,
```
Comment (kaushikb11) [inline flash/core/classification.py:44]:
```suggestion
        metrics: Optional[pl.metrics.Metric, Mapping, Sequence] = None,
```

**Comment by aribornstein** [comment]
Needs support for one hot embeddings

```python
        self.labels = [self.df[self.df["StudyInstanceUID"] == i].values.tolist()[0][1:-1] for i in self.ids]
```

**Comment by justusschock** [comment]
@kaushikb11 will do!

Currently this is just to get @aribornstein unblocked :)

**Comment by aribornstein** [comment]
Predict seems to still be categorical

**Comment by aribornstein** [comment]
This PR should address the categorical predict https://github.com/PyTorchLightning/lightning-flash/pull/156

File: flash/core/classification.py [hunk 2]
```diff
@@ -32,6 +35,79 @@ def after_uncollate(self, samples: Any) -> Any:
...
+    def predict(
+        self,
+        x: Any,
+        batch_idx: Optional[int] = None,
+        skip_collate_fn: bool = False,
+        dataloader_idx: Optional[int] = None,
+        data_pipeline: Optional[ClassificationDataPipeline] = None,
+    ) -> Any:    
+        if self.multilabel:
+            bind_method(self.data_pipeline,
```
Comment (aribornstein) [inline flash/core/classification.py:95]: I think this breaks if someone provides a data_pipeline as an argument but it works on the default pipeline.

**Comment by aribornstein** [comment]
@justusschock I think @tchaton broke this accidently

File: flash/data/data_utils.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import pandas as pd
 
 
+
 def labels_from_categorical_csv(
```
Comment (aribornstein) [inline flash/data/data_utils.py:7]: what is this called categorical_csv didn't we fix this

Comment (aribornstein) [reply flash/data/data_utils.py:7]: can someone give me write access on the branch it's very frustrating to have to submit pull requests from a fork every time

File: tests/vision/classification/test_data.py [hunk 4]
```diff
@@ -19,7 +19,7 @@
 from PIL import Image
 from torchvision import transforms as T
 
-from flash.data.data_utils import labels_from_categorical_csv
+from flash.data.data_utils import labels_from_csv
```
Comment (aribornstein) [inline tests/vision/classification/test_data.py:22]: See we renamed this why was is changed.

**Review by tchaton — COMMENTED** [review]
Any updates on this PR ?

**Comment by aribornstein** [comment]
It needs tests?

**Review by edgarriba — APPROVED** [review]
LGTM as soon conflicts are resolved. Which PR is blocking this one ?

**Pull request closed without merging**